tests/x11regressions/gnomecase/gnome_default_applications.pm

```diff
@@ -19,55 +19,82 @@ sub run {
     # Prepare test files
     x11_start_program("xterm");
     assert_screen 'xterm';
+    if (
+        check_default_apps(
+            # mimetype, application (to confirm if opens in the right application)
+            ['image/jpg',           'eog.desktop'],
+            ['image/png',           'eog.desktop'],
+            ['application/pdf',     'evince.desktop'],
+            ['application/x-bzip2', 'org.gnome.FileRoller.desktop'],
+            ['application/gzip',    'org.gnome.FileRoller.desktop']))
```

Comment:
IMHO that's not very readable. How about you call that before the if, assign it to a boolean value and check that?